report/3.Theory/Architecture.tex

```diff
@@ -0,0 +1,29 @@
+\section{System Architecture}
+Before the implementation and building phases can begin, it is essential to give some thought towards the design of the system, both software and hardware wise.
+In order to do so, a component architecture diagram was constructed, which can be seen on Figure~\ref{fig:comparch}.
+%TODO Insert figure
+
+As seen on the diagram, the system will consist of two platforms: the LEGO NXT for handling the real time handling, and a Raspberry Pi for handling the machine intelligence.
+Each platform is split into two groups, a hardware group consisting of whatever hardware is connected to it and a software group which contains a series of classes that must be implemented for the device to function.
```

Comment:
Whatever hardware is connected to it? What is it?
Also I believe some commas are lacking here.